xsl/openrosa2html5form.xsl

```diff
@@ -496,6 +495,49 @@ XSLT Stylesheet that transforms OpenRosa style (X)Forms into valid HTMl5 forms
         </xsl:if>
     </xsl:template>
 
+    <xsl:template match="*" mode="range-option-picker">
+        <xsl:param name="start" />
+        <xsl:param name="end" />
+        <xsl:param name="step" />
+        <xsl:param name="readonly" />
+        <xsl:if test="not($start > $end)">
+            <option>
+                <xsl:if test="$readonly">
+                    <xsl:attribute name="disabled">
+                        <xsl:value-of select="'disabled'"/>
+                    </xsl:attribute>
+                </xsl:if>
+                <xsl:attribute name="value">
+                    <xsl:choose>
+                        <xsl:when test="string($start)">
+                            <xsl:value-of select="$start" />
+                        </xsl:when>
+                        <xsl:otherwise>
+                            <xsl:message>ERROR: Could not determine value of select option.</xsl:message>
+                        </xsl:otherwise>
+                    </xsl:choose>
+                </xsl:attribute>
+                <xsl:value-of select="$start" />
+            </option>
+            <span>
+                <xsl:attribute name="data-option-value">
+                    <xsl:value-of select="$start" />
+                </xsl:attribute>
+                <xsl:attribute name="lang">
+                    <xsl:value-of select="@lang" />
+                </xsl:attribute>
+                <xsl:value-of select="."/>
+            </span>
+            <xsl:variable name="next" select="$start + $step"/>
+            <xsl:apply-templates select="current()" mode="range-option-picker">
+                <xsl:with-param name="start" select="$next"/>
```

Comment:
Nice solution. I think the only thing we need is some protection against overloading the server. The transformation of a long form can already bring down the server because it is so CPU intensive. So let's do a max of 500 iterations per range-picker question. What do you think about this?

Reply:
Agree, recursive cost a lot of memory. Code updated.